test/unittest/tests/test_prepare_mod_sort.cpp

```diff
@@ -113,14 +113,14 @@ init_op(WT_TXN_OP *op, WT_BTREE *btree, WT_TXN_TYPE type, uint64_t recno, WT_ITE
 
 /* Initialize a row-store key. */
 void
-init_key(WT_ITEM *key, const char *key_data)
+init_key(WT_SESSION_IMPL *session, WT_ITEM *key, std::string key_str)
 {
-    key->data = key_data;
-    key->size = sizeof(key_data);
+    WT_ASSERT(session, __wt_buf_init(session, key, key_str.size()) == 0);
+    WT_ASSERT(session, __wt_buf_set(session, key, key_str.c_str(), key_str.size()) == 0);
```

Comment:
I don't know about unittest, but in core WT code, WT_ASSERT is only compiled in for diagnostic mode (so `__wt_buf_init`, `__wt_buf_set` calls would not be called for release builds).  Regardless of how unittest is built, it would be good practice to follow the same guidelines - for each call, make the call, saving the return value, then use WT_ASSERT only on the return value.

Reply:
This is a good call - thanks for the feedback !